src/server/zset_family.cc

```diff
@@ -1286,6 +1286,12 @@ ScoredArray OpBZPop(Transaction* t, EngineShard* shard, std::string_view key, bo
 
   it_res->post_updater.Run();
 
+  auto res = iv.PopResult();
+
+  // We don't store empty keys
+  CHECK(!res.empty()) << key << " failed to pop from type " << pv.GetRobjWrapper()->encoding()
```

Comment:
also check fail at line 1273 that the value is not empty

Reply:
this line prints the size so it would've been irrelevant